lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -780,10 +780,13 @@ namespace {
             return Space();
           }
           
-          SmallVector<Space, 4> constrSpaces;
           bool foundBad = false;
           auto i = this->getSpaces().begin();
           auto j = other.getSpaces().begin();
+
+          SmallVector<Space, 4> copyParams(this->getSpaces().begin(),
```

Comment:
Some comment here, maybe instead of the one on 806. Something like "allocate space to hold the difference of the two spaces", but clearer, I hope. Better yet, maybe the whole for loop needs a comment explaining the algorithm.